### ISSUE TITLE
Get extraAnnotations from `KuberhealthyCheck` and set it to the pod that is actually for checker

### ISSUE THREAD

**Issue by Sn0rt** [Comcast/kuberhealthy#794]
1. set `extraAnnotations` for KuberhealthyCheck

I have set the extra annotation with  KuberhealthyCheck resource

```yaml
# kubectl  get khchecks.comcast.github.io daemonset -o yaml
apiVersion: comcast.github.io/v1
kind: KuberhealthyCheck
metadata:
  annotations:
  name: daemonset
  namespace: kuberhealthy
spec:
  extraAnnotations:
    v2-subnet: 7832d617-23ea-42e6-a5a3-efa92435a7c8
    v2-tenant: dcf722f63d0249f8b154876a79e1ce05
    v2-vpc: f203cacb-d15d-473d-9092-48c809f9f618
  podSpec:
    containers:
    - env:
      - name: POD_NAMESPACE
        valueFrom:
          fieldRef:
            fieldPath: metadata.namespace
      - name: CHECK_POD_TIMEOUT
        value: 10m
      image: kuberhealthy/daemonset-check:v3.2.3
      imagePullPolicy: IfNotPresent
      name: main
      resources:
        requests:
          cpu: 10m
          memory: 50Mi
      securityContext:
        allowPrivilegeEscalation: false
        readOnlyRootFilesystem: true
    securityContext:
      fsGroup: 999
      runAsUser: 999
    serviceAccountName: daemonset-khcheck
  runInterval: 15m
  timeout: 12m
```

2. get pod 

get the pod YAML by the command, `# kubectl  get pod daemonset-1611628373 -o yaml`

```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    comcast.github.io/check-name: daemonset
    v2-subnet: 7832d617-23ea-42e6-a5a3-efa92435a7c8
    v2-tenant: dcf722f63d0249f8b154876a79e1ce05
    v2-vpc: f203cacb-d15d-473d-9092-48c809f9f618
  creationTimestamp: "2021-01-26T02:32:58Z"
  labels:
    app: kuberhealthy-check
    kuberhealthy-check-name: daemonset
    kuberhealthy-run-id: 77a3f35e-c6bf-4368-9f7d-ec329b66a0e0
  name: daemonset-1611628373
  namespace: kuberhealthy
  ownerReferences:
  - apiVersion: v1
    kind: Pod
    name: kuberhealthy-54cd797c66-bzkh9
    uid: d78cd96e-1706-468c-a458-7131abc89756
  resourceVersion: "47152547"
  selfLink: /api/v1/namespaces/kuberhealthy/pods/daemonset-1611628373
  uid: 98d246b7-e816-41c0-a927-1f2afe14f85a
spec:
  containers:
  - env:
    - name: POD_NAMESPACE
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.namespace
    - name: CHECK_POD_TIMEOUT
      value: 10m
    - name: KH_REPORTING_URL
      value: http://kuberhealthy.kuberhealthy.svc.cluster.local/externalCheckStatus
    - name: KH_RUN_UUID
      value: 77a3f35e-c6bf-4368-9f7d-ec329b66a0e0
    - name: KH_CHECK_RUN_DEADLINE
      value: "1611629093"
    - name: KH_POD_NAMESPACE
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.namespace
    image: kuberhealthy/daemonset-check:v3.2.3
    imagePullPolicy: IfNotPresent
    name: main
    resources:
      requests:
        cpu: 10m
        memory: 50Mi
    securityContext:
      allowPrivilegeEscalation: false
      readOnlyRootFilesystem: true
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: daemonset-khcheck-token-qlfpt
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  nodeName: 10.90.66.135
  restartPolicy: Never
  schedulerName: default-scheduler
  securityContext:
    fsGroup: 999
    runAsUser: 999
  serviceAccount: daemonset-khcheck
  serviceAccountName: daemonset-khcheck
  terminationGracePeriodSeconds: 30

```

however, the actual pod to run not set the specific annotations. 

`kubectl  get pod daemonset-daemonset-1611629273-1611629284-xjv5w -o yaml`

```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    cluster-autoscaler.kubernetes.io/safe-to-evict: "true"
  creationTimestamp: "2021-01-26T02:48:04Z"
  generateName: daemonset-daemonset-1611629273-1611629284-
  labels:
    checkRunTime: "1611629284"
    controller-revision-hash: 77459b988
    creatingInstance: daemonset-1611629273
    kh-app: daemonset-daemonset-1611629273-1611629284
    khcheck: daemonset
    pod-template-generation: "1"
    source: kuberhealthy
  name: daemonset-daemonset-1611629273-1611629284-xjv5w
  namespace: kuberhealthy
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: DaemonSet
    name: daemonset-daemonset-1611629273-1611629284
    uid: 27a29f78-d1ec-4391-b8b9-e08bbc259d13
spec:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchFields:
          - key: metadata.name
            operator: In
            values:
            - 10.90.66.134
  containers:
  - image: gcr.io/google-containers/pause:3.1
    imagePullPolicy: IfNotPresent
    name: sleep
    resources:
      requests:
        cpu: "0"
        memory: "0"
    securityContext:
      runAsUser: 999
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-jll8j
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  nodeName: 10.90.66.134
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: default
  serviceAccountName: default
  terminationGracePeriodSeconds: 1
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/disk-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/memory-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/pid-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/unschedulable
    operator: Exists
  volumes:
  - name: default-token-jll8j
    secret:
      defaultMode: 420
      secretName: default-token-jll8j
```

3. expected 

The semantics of the ExtraAnnotations field may be ambiguous

```go
type CheckConfig struct {
	RunInterval      string            `json:"runInterval"`      // the interval at which the check runs
	Timeout          string            `json:"timeout"`          // the maximum time the pod is allowed to run before a failure is assumed
	PodSpec          apiv1.PodSpec     `json:"podSpec"`          // a spec for the external checker
	ExtraAnnotations map[string]string `json:"extraAnnotations"` // a map of extra annotations that will be applied to the pod
	ExtraLabels      map[string]string `json:"extraLabels"`      // a map of extra labels that will be applied to the pod
}
```

I hope that set the user-specific annotations

```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    cluster-autoscaler.kubernetes.io/safe-to-evict: "true"
    v2-subnet: 7832d617-23ea-42e6-a5a3-efa92435a7c8  // add this !!!
    v2-tenant: dcf722f63d0249f8b154876a79e1ce05 // add this !!
    v2-vpc: f203cacb-d15d-473d-9092-48c809f9f618 // add this !!
  creationTimestamp: "2021-01-26T02:48:04Z"
  generateName: daemonset-daemonset-1611629273-1611629284-
  labels:
    checkRunTime: "1611629284"
    controller-revision-hash: 77459b988
    creatingInstance: daemonset-1611629273
    kh-app: daemonset-daemonset-1611629273-1611629284
    khcheck: daemonset
    pod-template-generation: "1"
    source: kuberhealthy
  name: daemonset-daemonset-1611629273-1611629284-xjv5w
  namespace: kuberhealthy
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: DaemonSet
    name: daemonset-daemonset-1611629273-1611629284
    uid: 27a29f78-d1ec-4391-b8b9-e08bbc259d13
spec:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchFields:
          - key: metadata.name
            operator: In
            values:
            - 10.90.66.134
  containers:
  - image: gcr.io/google-containers/pause:3.1
    imagePullPolicy: IfNotPresent
    name: sleep
    resources:
      requests:
        cpu: "0"
        memory: "0"
    securityContext:
      runAsUser: 999
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-jll8j
      readOnly: true
```

**Comment by joshulyne** [comment]
@Sn0rt i believe you've caught on what happens when you set the extra annotations on the khcheck spec, the daemonset-check pod does get the same extra annotations, but the actual daemonset and daemonset pods don't. I believe if we wanted to implement this, user set annotations would have to be passed as an env-var on the khcheck instead of us assuming that all extra_annotations set in the khcheck spec are to be passed on to the daemonset / daemonset pods -- unless we do want to assume that? thoughts @integrii? . So for example:

```
apiVersion: comcast.github.io/v1
kind: KuberhealthyCheck
metadata:
  annotations:
  name: daemonset
  namespace: kuberhealthy
spec:
  podSpec:
    containers:
    - env:
      - name: POD_NAMESPACE
        valueFrom:
          fieldRef:
            fieldPath: metadata.namespace
      - name: CHECK_POD_TIMEOUT
        value: 10m
      - name: EXTRA_ANNOTATIONS
        value: ["v2-subnet: 7832d617-23ea-42e6-a5a3-efa92435a7c8", "v2-tenant: dcf722f63d0249f8b154876a79e1ce05", "v2-vpc: f203cacb-d15d-473d-9092-48c809f9f618"]
        *** OR can we use the downward API for this, like how we did for POD_NAMESPACE? ***
      image: kuberhealthy/daemonset-check:v3.2.3
      imagePullPolicy: IfNotPresent
      name: main
      resources:
        requests:
          cpu: 10m
          memory: 50Mi
      securityContext:
        allowPrivilegeEscalation: false
        readOnlyRootFilesystem: true
    securityContext:
      fsGroup: 999
      runAsUser: 999
    serviceAccountName: daemonset-khcheck
  runInterval: 15m
  timeout: 12m
```

**Comment by integrii** [comment]
I don't think we want to pass extra annotations from the `khcheck` resource.  I think this should be directly configurable on the daemonset specs.  It would be hard to assume that any extraAnnotations on the `khcheck` should also be on all the daemonset pods that are created by the checker pod, but it would be safe to assume that any pod created by the daemonset checker pod should have annotations directly configured on the daemonset checker pod's options.

I took a look at PR #795 and it makes sense, but I am not convinced that we can assume all `extraAnnotations` fields on the main `kuberhealthy` pod should always be added to the daemonset check's pods it creates.

In summary, it seems like we should add a new option (env var) to the daemonset checker `khcheck` that appends a specific set of annotations to the daemonset spec it creates.
